R/get_recommended_interventions_higher_order_cost.R

```diff
@@ -1,99 +1,90 @@
-#' for the intervention group in the next stage to be as close to the outcome goal
-#' as possible. "Optimal" means we are calculating the optimal interventions at the
-#' end of the stage, using the idea from web appendix section 5.1 of Nevo et al.
-#'
-#' For now, this defaults to "planning". we can discuss whether we need
-#' both "Planning" and "Optimal".
+#' @param intervention_lower_bounds numeric vector: lower bounds of the intervention components.
+#' For example: for a two-component intervention package, lower bounds could be
+#' c(0,0).
+#' @param intervention_upper_bounds numeric vector: upper bounds of the intervention components.
+#' For example: for a two-component intervention package, upper bounds could be
+#' c(10,20).
+#' @param outcome_goal numeric: the outcome goal, a desired probability or mean
+#' value.
+#' @param outcome_goal_opt_method character: the method used for calculating the
+#' recommended interventions. Either "numerical" or "grid_search".
+#' @param grid_search_step_size numeric: step size for the grid search algorithm
+#' when calculating the recommended interventions.
 #'
 #' @return List(recommended interventions, outcome goal, and estimated outcome mean/probability
 #' for the intervention group in the next stage )
 #'
 #' @examples
+#' # Example for using the grid search
+#' get_recommended_interventions_higher_order_cost(
+#'   beta_vec = c(0.1, 0.3, 0.15),
+#'   cost_list_of_lists = list(c(0, 1, 2), c(0, 4)),
+#'   intervention_lower_bounds = c(0, 0),
+#'   intervention_upper_bounds = c(10, 20),
+#'   grid_search_step_size = c(2, 5),
+#'   outcome_goal = 0.8,
+#'   outcome_goal_opt_method = "grid_search",
+#' )
+#'
+#' # Example for using the numerical method
 #' get_recommended_interventions_higher_order_cost(
 #'   beta_vec = c(0.1, 0.3, 0.15),
-#'   cost_list_of_lists = list(c(1, 2), c(4)),
+#'   cost_list_of_lists = list(c(0, 1, 2), c(0, 4)),
 #'   intervention_lower_bounds = c(0, 0),
 #'   intervention_upper_bounds = c(10, 20),
+#'   grid_search_step_size = c(2, 5),
 #'   outcome_goal = 0.8,
-#'   outcome_goal_optimization = "grid_search",
+#'   outcome_goal_opt_method = "numerical"
 #' )
 #'
 #' @importFrom rje expit logit
 #' @import stats
 #' @importFrom NlcOptim solnl
 #'
 #' @export
-#
-#
+#'
+#'
 get_recommended_interventions_higher_order_cost <- function(beta_vec,
                                                             cost_list_of_lists,
                                                             intervention_lower_bounds,
                                                             intervention_upper_bounds,
                                                             outcome_goal,
-                                                            outcome_goal_optimization = "numerical",
-                                                            grid_search_step_size = 0.01,
+                                                            outcome_goal_opt_method,
```

Comment:
Why don't we write it out "outcome_goal_optimization_method" just to be clear? This is pretty trivial so don't worry about it if you find it unnecessary. I was reading over it and was pausing for a bit to think because I thought "opt" meant "optional" not "optimization"

Reply:
agreed